README_BRANCHING.md

```diff
@@ -0,0 +1,41 @@
+This document attempts to capture the details of our branching strategy
+for Puppet Server.  This information is up-to-date as of 2015-03-06.
+
+## THE MOST IMPORTANT THING TO KNOW
+
+tl;dr: we are putting a freeze on merging up changes from `stable` to `master`
+until Puppet Server 2.0 ships.
```

Comment:
> That sounds like a lot more work to me; we're not just talking about dependencies, we're talking about major code changes in the Puppet Server code base.

Cool, just was something I was curious about as the plan has been developing.  I wasn't sure that we were talking about major changes in behavior beyond the dependency changes.